src/js/configuration_panel/button-listeners.js

```diff
@@ -176,50 +250,75 @@ https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
                         initialInputData = initialConfiguration[isAxes ? "axes" : "buttons"][inputIndex];
 
                     /**
-                     * Enable the "Save Changes" button if dropdown option is changed.
-                     * Otherwise, disable the button.
+                     * Enable the "Save Changes" and "Discard Changes" buttons if
+                     * dropdown option is changed.
                      */
                     var actionDropdown = configurationMenu.querySelector("select"),
                         currentDropdownValue = actionDropdown.value,
                         initialDropdownValue = initialInputData[isStoredData ? "currentAction" : "defaultAction"] || "null";
                     if (currentDropdownValue !== initialDropdownValue) {
                         saveChangesButton.removeAttribute("disabled");
+                        discardButton.removeAttribute("disabled");
+
+                        /**
+                         * Store the unsaved changes to avoid loss of configuration if
+                         * the panel is closed temporarily.
+                         */
+                        that.storeUnsavedChanges();
```

Comment:
So the "unsaved changes" material would actually be a good candidate for a live relay, i.e. something that listens to model changes resulting from form updates and immediately saves them to Chrome Storage.  It would also need to listen to changes to Chrome Storage and relay those to the model.

I'm happy to defer on some of this, i.e. one pull to migrate to using view components, another to use fluid-binder, another to add model relay to/from Chrome Storage.